qiskit_experiments/library/characterization/cr_hamiltonian.py

```diff
@@ -0,0 +1,472 @@
+
+        Returns:
+            A circuit decomposition for the cross resonance to measure.
+        """
+        cr_gate_p = circuit.Gate("cr_tone_p", num_qubits=2, params=[duration])
+        cr_gate_m = circuit.Gate("cr_tone_m", num_qubits=2, params=[duration])
+
+        cr_circuit = QuantumCircuit()
+        cr_circuit.add_register(2)
+        cr_circuit.append(cr_gate_p, [0, 1])
+        cr_circuit.x(0)
+        cr_circuit.append(cr_gate_m, [0, 1])
+
+        opt = self.experiment_options
+        with pulse.build(backend, default_alignment="left", name="cr_p") as cross_resonance_p:
+            # add cross resonance tone
+            pulse.play(
+                pulse.GaussianSquare(
+                    duration,
+                    amp=opt.amp,
+                    sigma=sigma,
+                    risefall_sigma_ratio=opt.risefall,
+                ),
+                pulse.control_channels(*self.physical_qubits)[0],
+            )
+            # add cancellation tone
+            if not np.isclose(opt.amp_t, 0.0):
+                pulse.play(
+                    pulse.GaussianSquare(
+                        duration,
+                        amp=opt.amp_t,
+                        sigma=sigma,
+                        risefall_sigma_ratio=opt.risefall,
+                    ),
+                    pulse.drive_channel(self.physical_qubits[1]),
+                )
+            else:
+                pulse.delay(duration, pulse.drive_channel(self.physical_qubits[1]))
+
+            # place holder for empty drive channels. this is necessary due to known pulse gate bug.
+            pulse.delay(duration, pulse.drive_channel(self.physical_qubits[0]))
+
+        with pulse.build(backend, default_alignment="left", name="cr_m") as cross_resonance_m:
+            with pulse.phase_offset(
+                np.pi,
+                pulse.control_channels(*self.physical_qubits)[0],
+                pulse.drive_channel(self.physical_qubits[1]),
+            ):
+                # add cross resonance tone
+                pulse.play(
+                    pulse.GaussianSquare(
+                        duration,
+                        amp=opt.amp,
+                        sigma=sigma,
+                        risefall_sigma_ratio=opt.risefall,
+                    ),
+                    pulse.control_channels(*self.physical_qubits)[0],
+                )
+                # add cancellation tone
+                if not np.isclose(opt.amp_t, 0.0):
+                    pulse.play(
+                        pulse.GaussianSquare(
+                            duration,
+                            amp=opt.amp_t,
+                            sigma=sigma,
+                            risefall_sigma_ratio=opt.risefall,
+                        ),
+                        pulse.drive_channel(self.physical_qubits[1]),
+                    )
+                else:
+                    pulse.delay(duration, pulse.drive_channel(self.physical_qubits[1]))
```

Comment:
This could be made more compact. Perhaps by using `Call` or defining the pulses once?